rmw_zenoh_cpp/src/detail/rmw_client_data.cpp

```diff
@@ -233,15 +233,21 @@ std::shared_ptr<ClientData> ClientData::make(
     return nullptr;
   }
 
-  auto client_data = std::shared_ptr<ClientData>(
-    new ClientData{
-      node,
-      std::move(entity),
-      request_members,
-      response_members,
-      std::move(request_type_support),
-      std::move(response_type_support)
-    });
+  std::lock_guard<std::mutex> lock(num_in_flight_mutex);
+  std::vector<std::shared_ptr<ClientData>> duplicate_pointers;
+  std::shared_ptr<ClientData> client_data;
+  do {
+    client_data = std::shared_ptr<ClientData>(
+      new ClientData{
+        node,
+        std::move(entity),
+        request_members,
+        response_members,
+        std::move(request_type_support),
+        std::move(response_type_support)
```

Comment:
If the first pointer is a duplicate, creating the second instance will segfault since some of the data members have been moved into the first one.
```suggestion
        entity,
        request_members,
        response_members,
        request_type_support,
        response_type_support
```